packages/cli-hermes/src/profileHermes/downloadProfile.ts

```diff
@@ -0,0 +1,175 @@
+ * @param ctx
+ * @param dstPath
+ * @param fileName
+ * @param sourceMapPath
+ * @param raw
+ * @param generateSourceMap
+ */
+export async function downloadProfile(
+  ctx: Config,
+  dstPath: string,
+  fileName?: string,
+  sourcemapPath?: string,
+  raw?: boolean,
+  shouldGenerateSourcemap?: boolean,
+) {
+  try {
+    const packageName = getPackageName(ctx);
+
+    // if file name is not specified, pull the latest file from device
+    const file = fileName || (await getLatestFile(packageName));
+    if (!file) {
+      logger.error(
+        'There is no file in the cache/ directory. Did you record a profile from the developer menu?\n',
+      );
+      process.exit(1);
+    }
+
+    logger.info(`File to be pulled: ${file}\n`);
+
+    //if destination path is not specified, pull to the current directory
+    dstPath = dstPath || ctx.root;
+
+    if (logger.isVerbose()) {
+      logger.info('Internal commands run to pull the file:\n');
+    }
+
+    //Copy the file from device's data to sdcard, then pull the file to a temp directory
+    execSyncWithLog(`adb shell run-as ${packageName} cp cache/${file} /sdcard`);
+
+    //If --raw, pull the hermes profile to dstPath
+    if (raw) {
+      execSyncWithLog(`adb pull /sdcard/${file} ${dstPath}`);
+      logger.success(`Successfully pulled the file to ${dstPath}/${file}\n`);
+    }
+
+    //Else: transform the profile to Chrome format and pull it to dstPath
+    else {
+      const osTmpDir = os.tmpdir();
+      const tempFilePath = path.join(osTmpDir, file);
+
+      execSyncWithLog(`adb pull /sdcard/${file} ${tempFilePath}`);
+
+      //If path to source map is not given
+      if (!sourcemapPath) {
+        //Get or generate the source map
+        if (shouldGenerateSourcemap) {
+          sourcemapPath = await generateSourcemap();
+        } else {
+          sourcemapPath = await findSourcemap(ctx);
+        }
+
+        //Run without source map
+        if (!sourcemapPath) {
+          logger.warn(
+            'Cannot generate or find bundle and source map, running the transformer without source map\n',
+          );
+          logger.info(
+            'Instructions on how to get source map: set `bundleInDebug: true` in your app/build.gradle file, inside the `project.ext.react` map.\n',
+          );
+        }
+      }
+
+      //Run transformer tool to convert from Hermes to Chrome format
+      const events = await transformer(
+        tempFilePath,
+        sourcemapPath,
+        'index.bundle',
+      );
+
+      const transformedFilePath = `${dstPath}/${path.basename(
+        file,
+        '.cpuprofile',
+      )}-converted.json`;
+      fs.writeFileSync(
+        transformedFilePath,
+        JSON.stringify(events, undefined, 4),
+        'utf-8',
+      );
+      logger.success(
+        `Successfully converted to Chrome tracing format and pulled the file to ${transformedFilePath}\n`,
+      );
+    }
+  } catch (e) {
+    throw new Error(e.message);
```

Comment:
You may consider providing more descriptive error message and passing the original error to the CLIError to allow CLI to handle it using its standard error handling mechanism.

Example:
https://github.com/react-native-community/cli/blob/79498d598e00e481e6058a2892eaa71870215922/packages/platform-android/src/commands/runAndroid/tryLaunchAppOnDevice.ts#L41